src/api/Interpret.cc

```diff
@@ -164,6 +171,11 @@ void Interpret::setOption(ASTNode& n) {
     else // Normal option
         name = strdup(n.getValue());
 
+    if (logic && isPreLogicOption(name)) {
```

Comment:
Would it make sense to move this into SMTConfig?  We'd need to update the logic in the config file so that the config knows whether it has been set.  This could break some binaries linking to osmt possibly.

Reply:
We actually have in place a method of keeping the solver's logic and config's logic in sync. This code could fit well with  PR #139.

Reply:
Changes related interpolation interface has been removed from this PR and a separate PR will be created for that.